inc/Engine/Preload/Controller/CheckExcludedTrait.php

```diff
@@ -14,4 +14,33 @@ protected function is_excluded( string $url ) {
 		$excluded = str_replace( '/', '\/', get_rocket_cache_reject_uri() );
 		return (bool) preg_match( "/$excluded/", $url );
 	}
+
+	/**
+	 * Check if the url is excluded by using a filter.
+	 *
+	 * @param string $url url to check.
+	 * @return bool
+	 */
+	protected function is_excluded_by_filter( string $url ): bool {
+		/**
+		 * Regex to exclude URI from preload.
+		 *
+		 * @param string[] regexes to check
+		 */
+		$regexes = (array) apply_filters( 'rocket_preload_exclude_urls', [] );
+
+		if ( empty( $regexes ) ) {
+			return false;
+		}
+
```

Comment:
can we plz make this array unique, we are adding into this filter from two places so we need to make sure that it's unique  and no duplication.
something like
```
$regexes = array_flip( array_flip( $regexes ) );
```